src/components/Template.js

```diff
@@ -137,4 +137,15 @@ function transformHelpersToHogan(helpers, compileOptions, data) {
   );
 }
 
-export default Template;
+// Resolve transformData before Template, so transformData is always called
+// even if the data is the same. Allowing you to dynamically inject conditions in
+// transformData that will force re-rendering
+const withTransformData =
```

Comment:
Have you considered using alternative implementations? What about use transformData in willReceiveProps and store the result in the state?